jdaviz/core/template_mixin.py

```diff
@@ -1677,6 +1679,69 @@ def _check_dataset_spectral_subset_valid(self, event={}, return_ranges=False):
             return self.spectral_subset_valid
 
 
+class NonFiniteUncertaintyMismatchMixin(VuetifyTemplate, HubListener):
+    """ Adds a traitlet that identifies if there are any finite data values
+        that correspond to a non-finite uncertainty at that index.
+
+        In model fitting, the presence of finite, fittable data with corresponding
+        non-finite uncertainties can cause issues. Finite data values will be
+        filtered out in this case which may be undesirable. This traitlet when
+        True triggers a warning in the model fitting plugin (in specviz only,
+        currently) if there are any finite values with non-finite uncertainties.
+
+        Note that if a the uncertainty array is FULLY non-finite and the data is
+        FULLY finite, uncertainties will be set to None (in the Specviz parser),
+        so this traitlet will be False in that case (and therefore no warning
+        message displayed in the plugin).
+    """
+
+    non_finite_uncertainty_mismatch = Bool(False).tag(sync=True)
+
+    # every time a data/subset selection is changed, check the data selection and
+    # its uncertainties to see if there are any finite data elements with
+    # uncertainties. Warn in plugin if this occurs.
+
+    # note! the 'n/a' traitlet doesn't exist. we are watching it here because
+    # DatasetSpectralSubsetValidMixin is also watching the exact set
+    # of real traitlets, which creates a bug where only the last inhereted method
+    # will run, so we need to spoof it like this.
+    @observe("n/a", "dataset_selected", "spectral_subset_selected")
```

Comment:
Are you sure observing "n/a" is necessary? I don't understand what you mean by "watching the exact set of real traitlets", maybe someone can explain it to me if I'm missing something?

Reply:
Just observing "dataset_selected" and "spectral_subset_selected" would conflict with another observe and only the last one added would actually be called.  @cshanahan1 - is there an issue for this yet that we can reference?

Reply:
will post the issue in a sec, i was getting an example code snippet working to create the issue

Reply:
Has this been resolved? I also don't understand what "n/a" is for.

Reply:
i think, somehow miraculously, im not seeing this anymore so i've removed the dummy traitlet.